scripts/sonar-analysis.sh

```diff
@@ -0,0 +1,62 @@
+#!/bin/bash
+
+set -euo pipefail
+
+# Fetch all commit history so that SonarQube has exact blame information
+# for issue auto-assignment
+# This command can fail with "fatal: --unshallow on a complete repository does not make sense"
+# if there are not enough commits in the Git repository
+# For this reason errors are ignored with "|| true"
+git fetch --unshallow || true
+
+# fetch references from github for PR analysis
+if [ -n "${GITHUB_BASE_BRANCH}" ]; then
+	git fetch origin "${GITHUB_BASE_BRANCH}"
+fi
+
+# PIPELINE_ID is used by burgr to identify stages of the pipeline
+if [ -z "$PIPELINE_ID" ]; then
+  PIPELINE_ID=$BUILD_NUMBER
+fi
+
+if [ "$PULL_REQUEST" != "false" ]; then
+  echo "======= Analyze PR $PULL_REQUEST"
+  node_modules/sonarqube-scanner/src/bin/sonar-scanner \
```

Comment:
npx sonarqube-scanner ?
https://www.npmjs.com/package/sonarqube-scanner
or a gulp.task for consistency?
https://www.npmjs.com/package/gulp-sonar